### ISSUE TITLE
"x in int()" is not flagged as an error

### ISSUE THREAD

**Issue by JelleZijlstra** [quora/pyanalyze#475]
Test case:
```python
def capybara(y: int):
    reveal_type(1 in y)
```

**Comment by JelleZijlstra** [comment]
Note that `in` has two steps:
- Call `__contains__`
- Iterate over the object (`__iter__`) and check all the elements
So we should allow `in` if the RHS has either of these methods. We should also type check the arguments (#476).
